tests/Unit/IO/Importers/Test_ReadSpecThirdOrderPiecewisePolynomial.cpp

```diff
@@ -0,0 +1,254 @@
+
+  test_options();
+
+  using MockRuntimeSystem = ActionTesting::MockRuntimeSystem<Metavariables>;
+
+  const size_t self_id{1};
+
+  // Create a temporary file with test data to read in
+  // First, check if the file exists, and delete it if so
+  const std::string test_filename{"TestSpecFuncOfTimeData.h5"};
+  constexpr uint32_t version_number = 4;
+  if (file_system::check_if_file_exists(test_filename)) {
+    file_system::rm(test_filename, true);
+  }
+
+  h5::H5File<h5::AccessType::ReadWrite> test_file(test_filename);
+
+  const std::array<double, 2> expected_times{{0.0, 0.00599999974179903}};
+  const std::array<std::string, 2> expected_names{
+      {"ExpansionFactor", "RotationAngle"}};
+
+  const std::vector<std::vector<double>> test_expansion{
+      {5.9999997417990300e-03, 0.0, 1.0, 3.0, 1.0, 1.0, -1.4268296756999999e-06,
+       0.0, 0.0},
+      {5.9999997417990300e-03, 5.9999997417990300e-03, 1.0, 3.0, 1.0,
+       9.9999999143902230e-01, -1.4268296756999999e-06, 0.0,
+       -3.5943676411727592e-05}};
+  const std::vector<std::string> expansion_legend{
+      "Time", "TLastUpdate", "Nc",  "DerivOrder", "Version",
+      "a",    "da",          "d2a", "d3a"};
+  auto& expansion_file = test_file.insert<h5::Dat>(
+      "/" + expected_names[0], expansion_legend, version_number);
+  expansion_file.append(test_expansion);
+
+  const std::vector<std::vector<double>> test_rotation{
+      {5.9999997417990300e-03, 0.0, 1.0, 3.0, 1.0, 0.0, 1.3472907726000001e-02,
+       0.0, 0.0},
+      {5.9999997417990300e-03, 5.9999997417990300e-03, 1.0, 3.0, 1.0,
+       8.0837442877282155e-05, 1.3472907726000001e-02, 0.0,
+       1.4799266358888008e-04}};
+  const std::vector<std::string> rotation_legend{
+      "Time", "TLastUpdate", "Nc",    "DerivOrder", "Version",
+      "Phi",  "dPhi",        "d2Phi", "d3Phi"};
+  auto& rotation_file = test_file.insert<h5::Dat>(
+      "/" + expected_names[1], rotation_legend, version_number);
+  rotation_file.append(test_rotation);
+
+  MockRuntimeSystem runner{{}};
+  std::unordered_map<std::string,
+                     std::unique_ptr<::domain::FunctionsOfTime::FunctionOfTime>>
+      initial_functions_of_time{};
+
+  const std::array<DataVector, 4> initial_coefficients{
+      {{0.0}, {0.0}, {0.0}, {0.0}}};
+  initial_functions_of_time["ExpansionFactor"] =
+      std::make_unique<domain::FunctionsOfTime::PiecewisePolynomial<3>>(
+          0.0, initial_coefficients);
+  initial_functions_of_time["RotationAngle"] =
+      std::make_unique<domain::FunctionsOfTime::PiecewisePolynomial<3>>(
+          0.0, initial_coefficients);
+
+  ActionTesting::emplace_component_and_initialize<component<Metavariables>>(
+      &runner, self_id,
+      {std::string{test_filename},
+       std::map<std::string, std::string>{
+           {"ExpansionFactor", "ExpansionFactor"},
+           {"RotationAngle", "RotationAngle"}},
+       std::move(initial_functions_of_time)});
+
+  runner.set_phase(Metavariables::Phase::Testing);
```

Comment:
we now have non-member functions `ActionTesting::set_phase` that are preferred, same with `next_action`